lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemLoader.java

```diff
@@ -181,6 +183,9 @@ private static S3ClientFactory s3ClientFactory(SdkHttpClient httpClient, OpenTel
             endpoint.map(URI::create).ifPresent(s3::endpointOverride);
             s3.forcePathStyle(pathStyleAccess);
 
+            s3.requestChecksumCalculation(RequestChecksumCalculation.WHEN_REQUIRED);
```

Comment:
add a comment why this change and maybe add an issue

Reply:
I will before pushing